### ISSUE TITLE
hot_threads API broken on 5.5.0

### ISSUE THREAD

**Issue by pemontto** [elastic/logstash#7608]
The hot_threads API seems to be broken on the latest 5.5.0 release.

- Version: 5.5.0
- Operating System: Centos7
- Config File (if you have sensitive info, please remove it): All
- Steps to Reproduce: `curl localhost:9603/_node/hot_threads`

```
{
  "status": 500,
  "request_method": "GET",
  "path_info": "/_node/hot_threads",
  "query_string": "pretty",
  "http_version": "HTTP/1.1",
  "http_accept": "*/*",
  "error": "Unexpected Internal Error",
  "class": "Java::JavaLang::IllegalStateException",
  "message": "Duplicate key {cpu.time=10462123727, waited.time=-1, thread.state=timed_waiting, waited.count=9692, thread.name=[main]<redis, blocked.count=92, thread.stacktrace=[java.lang.Object.wait(Native Method), org.jruby.RubyThread.sleep(RubyThread.java:1002), org.jruby.RubyKernel.sleep(RubyKernel.java:803)], blocked.time=-1}",
  "backtrace": [
    "java.util.stream.Collectors.lambda$throwingMerger$0(java/util/stream/Collectors.java:133)",
    "java.util.HashMap.merge(java/util/HashMap.java:1253)",
    "java.util.stream.Collectors.lambda$toMap$58(java/util/stream/Collectors.java:1320)",
    "java.util.stream.ReduceOps$3ReducingSink.accept(java/util/stream/ReduceOps.java:169)",
    "java.util.ArrayList$ArrayListSpliterator.forEachRemaining(java/util/ArrayList.java:1374)",
    "java.util.stream.AbstractPipeline.copyInto(java/util/stream/AbstractPipeline.java:481)",
    "java.util.stream.AbstractPipeline.wrapAndCopyInto(java/util/stream/AbstractPipeline.java:471)",
    "java.util.stream.ReduceOps$ReduceOp.evaluateSequential(java/util/stream/ReduceOps.java:708)",
    "java.util.stream.AbstractPipeline.evaluate(java/util/stream/AbstractPipeline.java:234)",
    "java.util.stream.ReferencePipeline.collect(java/util/stream/ReferencePipeline.java:499)",
    "org.logstash.instrument.reports.ThreadsReport.generate(org/logstash/instrument/reports/ThreadsReport.java:29)",
    "java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)",
    "RUBY.initialize(/usr/share/logstash/logstash-core/lib/logstash/api/commands/hot_threads_reporter.rb:11)",
    "RUBY.hot_threads(/usr/share/logstash/logstash-core/lib/logstash/api/commands/node.rb:58)",
    "RUBY.HEAD /hot_threads(/usr/share/logstash/logstash-core/lib/logstash/api/modules/node.rb:20)",
    "org.jruby.RubyMethod.call(org/jruby/RubyMethod.java:116)",
    "RUBY.compile!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1611)",
    "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)",
    "RUBY.route!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:975)",
    "RUBY.route_eval(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:994)",
    "RUBY.route!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:975)",
    "RUBY.process_route(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1015)",
    "org.jruby.RubyKernel.catch(org/jruby/RubyKernel.java:1242)",
    "RUBY.process_route(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1013)",
    "RUBY.route!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:973)",
    "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)",
    "RUBY.route!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:972)",
    "RUBY.dispatch!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1085)",
    "RUBY.invoke(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067)",
    "org.jruby.RubyKernel.catch(org/jruby/RubyKernel.java:1242)",
    "RUBY.invoke(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067)",
    "RUBY.dispatch!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1082)",
    "RUBY.call!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:907)",
    "RUBY.invoke(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067)",
    "org.jruby.RubyKernel.catch(org/jruby/RubyKernel.java:1242)",
    "RUBY.invoke(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067)",
    "RUBY.call!(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:907)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:895)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-protection-1.5.3/lib/rack/protection/xss_header.rb:18)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-protection-1.5.3/lib/rack/protection/path_traversal.rb:16)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-protection-1.5.3/lib/rack/protection/json_csrf.rb:18)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-protection-1.5.3/lib/rack/protection/frame_options.rb:31)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-1.6.6/lib/rack/nulllogger.rb:9)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-1.6.6/lib/rack/head.rb:13)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:182)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/sinatra-1.4.8/lib/sinatra/base.rb:2013)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-1.6.6/lib/rack/urlmap.rb:66)",
    "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-1.6.6/lib/rack/urlmap.rb:50)",
    "RUBY.call(/usr/share/logstash/logstash-core/lib/logstash/api/rack_app.rb:57)",
    "RUBY.call(/usr/share/logstash/logstash-core/lib/logstash/api/rack_app.rb:31)",
    "RUBY.call(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/rack-1.6.6/lib/rack/builder.rb:153)",
    "RUBY.handle_request(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/puma-2.16.0-java/lib/puma/server.rb:557)",
    "RUBY.process_client(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/puma-2.16.0-java/lib/puma/server.rb:404)",
    "RUBY.run(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/puma-2.16.0-java/lib/puma/server.rb:270)",
    "org.jruby.RubyProc.call(org/jruby/RubyProc.java:281)",
    "RUBY.spawn_thread(/usr/share/logstash/vendor/bundle/jruby/1.9/gems/puma-2.16.0-java/lib/puma/thread_pool.rb:106)",
    "java.lang.Thread.run(java/lang/Thread.java:745)"
  ]
}
```

**Comment by andrewvc** [comment]
@pemontto it seems like this bug is due to two threads having the same name. Somewhere in 5.5 we stopped limited the threads to just worker threads. I think this is a good change, even though its source isn't clear. The fix, I think, would be adding the `threadId` to the map of results.

**Comment by andrewvc** [comment]
@pemontto https://github.com/elastic/logstash/pull/7617 should address this once its finished and merged. I'm targetting next week for that. Thanks for reporting this important bug.

**Comment by genuss** [comment]
As #7790 is not released, does anyone knows a workaround?

**Comment by genuss** [comment]
It's released now in 5.5.2. Works like a charm! Thanks a lot for the fix!
